wordpress-mobile/WordPress-Android
# Put reader EventBus behind feature flag

Depends on #11193 that must be merged first.

This is related to this PR [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/11193#issuecomment-582372512). 

We want to register/unregister the EventBus in the `ReaderPostListViewModel` only if we have `INFORMATION_ARCHTECTURE_AVAILBALE` feature flag active. We actually should not have issues even with the code as it is now but better to be explicit on the intended usage.

## To test


### Test case 1 - reproduce
- Compile and run vanilla flavor before this patch
- Navigate into the reader
- Check logcat and see you have `READER: Subfilter bottom sheet > followed tags changed` and/or `READER: Subfilter bottom sheet > followed blogs changed` logs (alternatively put a breakpoint in the 2 `ReaderPostListViewModel` > `fun onEventMainThread` functions and check that at least one of them is invoked)

### Test case 2 - check the fix
- Compile and run vanilla flavor with this PR patch
- Navigate into the reader
- Check logcat and see you DO NOT have `READER: Subfilter bottom sheet > followed tags changed` and/or `READER: Subfilter bottom sheet > followed blogs changed` logs (alternatively put a breakpoint in the 2 `ReaderPostListViewModel` > `fun onEventMainThread` functions and check that none of them is invoked)

PR submission checklist:
